src/MSBuildLocator/DotNetSdkLocationHelper.cs

```diff
@@ -111,8 +124,42 @@ private static string GetDotNetBasePath(string workingDirectory)
             {
                 return null;
             }
+            
+            var basePath = matched.Groups[1].Value.Trim();
```

Comment:
nit: `GetDotNetBasePaths` could be an iterator method and simply `yield return basePath` here and then `yield return path` in the loop below. Just a thought, looks good either way.

Reply:
I like this idea and have submitted a new commit with this change. Looks quite nice and clean.